src/frontend/src/scheduler/snapshot.rs

```diff
@@ -42,47 +45,60 @@ pub enum ReadSnapshot {
     /// A frontend-pinned snapshot.
     FrontendPinned {
         snapshot: PinnedSnapshotRef,
-        // It's embedded here because we always use it together with snapshot.
-        is_barrier_read: bool,
     },
 
+    BarrierRead,
```

Comment:
Given that we switch to use MAX epoch here, I suggest we rename it to `ReadUncommitted`